### PR TITLE
fix: fix flaky feature gate tests

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
fix: fix flaky feature gate tests
When user disables deploy feature gate, tests will wait until all resources are deleted. This is wrong, beceause tto will not delete any resources, but it will just disable reconciliation. This patch updates behaviour of the test, to try if the operator is still reconciling the resources

Signed-off-by: Karel Šimon <ksimon@redhat.com>

**Release note**:
```
NONE
```

### CODE REVIEW

File: tests/tekton-tasks_test.go
Comment (0xFelix) [inline tests/tekton-tasks_test.go:81]: Wouldn't `Consistently` be more appropriate for this test to verify that the operator does not create new objects anymore?

https://pkg.go.dev/github.com/onsi/gomega#Consistently

File: tests/tekton-tasks_test.go
Comment (0xFelix) [inline tests/tekton-tasks_test.go:107]: Same as above.

File: tests/tekton-tasks_test.go
Comment (0xFelix) [inline tests/tekton-tasks_test.go:133]: Same as above.

File: tests/tekton-tasks_test.go
Comment (0xFelix) [inline tests/tekton-tasks_test.go:58]: IMO I prefer `errors.IsNotFound(err)` but I leave it up to you!

**Review by 0xFelix — APPROVED** [review]
Thanks!

/approve
/lgtm

Giving you time to react
/hold

File: tests/tekton-tasks_test.go
Comment (ksimon1) [reply tests/tekton-tasks_test.go:58]: In this case, it might be better to use `ReasonForError` because when the error is different, it will show you what typer of error it is (better for debugging). In `errors.IsNotFound(err)` it will show you just true / false.

**Pull request merged**
